File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let prev = {}
    
    for (let i = 0; i < nums.length; i ++) {
        const num = nums[i]
        const reminder = target - num
        if ((reminder in prev)) return [prev[reminder], i]
        prev[num] = i
    }
};

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        prevs = {}
        
        for i in range(len(nums)):
            num = nums[i]
            diff = target - num
            if num in prevs.keys():
                return [i, prevs[num]]
            prevs[diff] = i
        